src/SourceGenerators/Uno.UI.SourceGenerators/HotRestart/HotRestartGenerator.cs

```diff
@@ -0,0 +1,352 @@
+﻿#nullable enable
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Uno.UI.SourceGenerators.Helpers;
+using System.Diagnostics;
+using Uno.Extensions;
+using Uno.Roslyn;
+
+namespace Uno.UI.SourceGenerators.HotRestart
+{
+	[Generator]
+	public class HotRestartGenerator : ISourceGenerator
+	{
+		public void Initialize(GeneratorInitializationContext context)
+		{
+		}
+
+		public void Execute(GeneratorExecutionContext context)
+		{
+			if (!DesignTimeHelper.IsDesignTime(context)
+				&& PlatformHelper.IsIOS(context)
+				&& PlatformHelper.IsUnoHead(context))
+			{
+				var generator = new Generator(context);
+				generator.Run();
+			}
+		}
+
+		private class Generator
+		{
+			private readonly GeneratorExecutionContext _context;
+
+			private INamedTypeSymbol? _mauiAppDelegate;
+
+			public Generator(GeneratorExecutionContext context)
+			{
+				_context = context;
+			}
+
+			public void Run()
+			{
+				if (!IsGenerationEnabled())
+				{
+					return;
+				}
+
+				_mauiAppDelegate = _context.Compilation.GetTypeByMetadataName("Microsoft.Maui.MauiUIApplicationDelegate");
+
+				var builder = new IndentedStringBuilder();
+				builder.AppendLineIndented("#if DEBUG");
+				builder.AppendLineIndented("// <auto-generated>");
+				builder.AppendLineIndented("// *************************************************************");
+				builder.AppendLineIndented($"// This file has been generated by Uno.UI ({nameof(HotRestartGenerator)})");
+				builder.AppendLineIndented("// *************************************************************");
+				builder.AppendLineIndented("// </auto-generated>");
+				builder.AppendLine();
+				builder.AppendLineIndented("using System;");
+				builder.AppendLine();
+
+				TryGenerateMAUIAppDelegate(builder);
+				GenerateHotRestartDelegate(builder);
+				builder.AppendLineIndented("#endif");
```

Comment:
Almost all of the generated code is constant, except for `appType` in `GenerateHotRestartDelegate`. I wonder if we can just use an interpolated raw string literal for the code.

Reply:
That's fair. Indeed, most of it can be in a raw literal, likely adding an `#if false` for the `MauiUIApplicationDelegate` type, in case it exists.